File: DesignPatternsExercises/java/MementoDesignPattern/Exercise/HomeTheatre.java
public class HomeTheatre {
    private static int stateCount = 0;
    private LedTV ledTV;
    private Speaker speaker;

    public HomeTheatre(LedTV ledTV, Speaker speaker) {
        this.ledTV = ledTV;
        this.speaker = speaker;
        __________(1)__________++;
    }

    public static int getStateCount() {
        return stateCount;
    }

    public Memento getMemento() {
        return __________(2)__________(ledTV, speaker);
    }

    public void setMemento(Memento memento) {
        ledTV = __________(3)__________();
        speaker = __________(4)__________();
    }

    @Override
    public String toString() {
        return ledTV + "\n    " + speaker;
    }
}

File: DesignPatternsExercises/ruby/memento_design_pattern/exercise/home_theatre.rb
require_relative 'memento'

class HomeTheatre
  @@state_count = 0

  def initialize(led_tv, speaker)
    @led_tv = led_tv
    @speaker = speaker
    __________(1)__________ += 1
  end

  def __________(2)__________
    @@state_count
  end

  def memento
    __________(3)__________(@led_tv, @speaker)
  end

  def memento=(memento)
    @led_tv = __________(4)__________
    @speaker = __________(5)__________
  end

  def to_string
    "#{@led_tv.to_string}\n    #{@speaker.to_string}"
  end
end
